lib/league_stats.rb

```diff
@@ -0,0 +1,139 @@
+require 'csv'
+require_relative 'stats'
+require_relative 'game'
+require_relative 'team'
+require_relative 'game_teams'
+
+class LeagueStats < Stats
+  attr_reader :games, :teams, :game_teams
+
+  def initialize(data)
+    @games = build_games(data)
+    @teams = build_teams(data)
+    @game_teams = build_game_teams(data)
+  end
+
+  def count_of_teams
+    @teams.count
+  end
+
+  def best_offense
+    avg_goals = average_team_goals_all_seasons(team_goals_hash, games_per_team)
```

Comment:
Very Dry code using the average_team_goals_all_seasons helper method in league_stats.rb.

Reply:
Thank you!